wp/lib/bap_wp/src/constraint.ml

```diff
@@ -209,18 +209,43 @@ let rec eval_aux ?stats:(stats = init_stats) (constr : t) (olds : z3_expr list)
       (eval_aux ~stats:stats c2 olds news ctx)
   | Clause (hyps, concs) ->
     let eval_conjunction conj =
-      List.map conj ~f:(fun c -> eval_aux ~stats:stats c olds news ctx)
-      |> Bool.mk_and ctx
+      if List.length conj = 1 then 
+        eval_aux ~stats:stats (List.hd_exn conj) olds news ctx (* This is tail recursive. Avoids And node. *)
+      else
+        List.map conj ~f:(fun c -> eval_aux ~stats:stats c olds news ctx)
```

Comment:
Do you know if there are performance impacts if we create an `and` with an empty list?

Reply:
I think there were, but now I am not 100% sure.

Reply:
👍